### ISSUE TITLE
Stripe Checkout

### ISSUE THREAD

**Issue by philsward** [backdrop-contrib/uc_stripe#2]
Disclaimer: I have not tried this module to know it's full scope. I can only assume it maintains the same type of functionality as uc_stripe for Drupal 7.

Is there any interest in marrying the current implementation with [Stripe Checkout](https://stripe.com/payments/checkout)?

I had a variety of problems on the D7 uc_stripe so I had someone custom program me a new version that is based on the Stripe Checkout API. Personally, I love it.

If you aren't familiar, Stripe Checkout works a lot like PayPal where it re-directs the user to a Stripe hosted page to handle the CC info. After moving over to this, I have had a LOT less problems with customers cards not working, getting various errors during checkout etc.

I'm willing to share the D7 module as it sits now (it was last updated late Aug early Sept) for inclusion into uc_stripe. What I envision is to have an option to choose between the on-site CC collection or off-site (stripe checkout) CC collection and give the user the ability to choose between one or the other. This would reduce the clutter of having a bunch of similar modules out there that almost do the same thing, just a little different.

**Comment by laryn** [comment]
@philsward I'm open to the possibility. Feel free to post the version you have in a repo and link it here.

**Comment by philsward** [comment]
So... I'm not very good with git, but let's give this a go:

https://github.com/philsward/uc_stripecheckout/releases/tag/1.7

**Comment by laryn** [comment]
@philsward I'll try to take a look at this. Is it GPL-2.0 (compatible for inclusion with this module)? Is it a modified version of `uc_stripe` or something completely new?

**Comment by philsward** [comment]
I need to set the license for it... As far as I'm concerned (as the owner)
it's GPL 2.0

It may have some code that was pulled from uc_stripe but otherwise it is
its own module.

**Comment by laryn** [comment]
Okay -- just making sure we can include it from a licensing perspective. (I think if you've pulled a bunch of code from `uc_stripe` as a base the resulting code has to be compatible by default, but good to see a formal GPL-2.0 on it.)

Do you know what happens currently if you install this module alongside `uc_stripe`? (I see this in the readme: "uc_stripe and uc_stripecheckout cannot be used at the same time.")

**Comment by philsward** [comment]
I'll bring the licensing issue up with my developer and see what we need to do on that topic.

As far as running them both, I had both modules enabled on a live site for months and didn't notice any I'll effects. uc_stripe was not enabled as a payment method however. It was later the developer informed me they should not both be enabled at the same time.

I looked through our email thread for a reason, but never found a reason why.

During this discussion, I found customers who were having problems checking out and I think it was believed that the two co-existing was the reason but later realized it was a problem with how the page was redirecting to stripecheckout, so I'm honestly at a loss.

**Comment by philsward** [comment]
> I opened up an issue for Backdrop uc_stripe and opened it up for them to include your work with uc_stripecheckout as an alternative option for the payment method.

I would suggest porting uc_stripecheckout to Backdrop separately, and not marrying with uc_stripe.



 > 1) Are you OK with me opening your work to the public?

OK.



 > 2) Should we add any sort of license to the work?

From my side, no.



 > 3) There is a question about "why uc_stripe must be disabled" that I am unable to answer. If you have a quick response, I can update that issue and answer that question.

The uc_stripe does various things even when it is enabled as module, but not enabled as payment method. One of them is that it disables the "Review order" button on the checkout page by default and enables it via Javascript. I think uc_stripe checks only whether it is the default credit card gateway, but it does not check whether the credit card gateway is enabled as payment method at all. So to make sure that uc_stripe does not interfere with anything when it is not used, it is better to disable the uc_stripe module when uc_stripecheckout is used.
In other words, the uc_stripecheckout is a complete replacement for uc_stripe, so there is no advantage to keep uc_stripe module enabled, but there might be disadvantages.
